tooling/miniutil/src/run.rs

```diff
@@ -0,0 +1,25 @@
+use crate::*;
+
+use gen_minirust::prelude::NdResult;
+
+// Run the program and return its TerminationInfo.
+// We fix `BasicMemory` as a memory for now.
+pub fn run_program(prog: Program) -> TerminationInfo {
+    let res: NdResult<!> = try {
+        let mut machine = Machine::<BasicMemory>::new(prog)?;
+
+        loop {
+            machine.step()?;
+
+            // Drops everything not reachable from `machine`.
+            gen_minirust::libspecr::hidden::mark_and_sweep::<Machine<BasicMemory>>(machine);
```

Comment:
Do you have any clue on why Rust can't figure out this generic parameter itself?
Even  `let mut machine: Machine<BasicMemory>` doesn't help.

mark_and_sweep just get's the machine as `impl GcCompat`

Reply:
`impl GcCompat` can't even be specified via turbofish... can you link me to where the function is declared?

Reply:
Haha, I just confused myself with `fn mark_and_sweep<T: GcCompat>(root: impl GcCompat)` ^^
New version is on the way